File: opensource/test/src/main/java/com/sos/scheduler/engine/test/DelegatingSchedulerController.java
package com.sos.scheduler.engine.test;

import com.sos.scheduler.engine.eventbus.SchedulerEventBus;
import com.sos.scheduler.engine.kernel.settings.Settings;
import com.sos.scheduler.engine.kernel.util.Time;
import com.sos.scheduler.engine.main.SchedulerController;
import com.sos.scheduler.engine.main.SchedulerThreadController;

abstract class DelegatingSchedulerController implements SchedulerController {
    private final SchedulerThreadController delegate;

    protected DelegatingSchedulerController(String name) {
        delegate =  new SchedulerThreadController(name);
    }

    @Override public final void setSettings(Settings o) {
        delegate.setSettings(o);
    }

    @Override public final Settings getSettings() {
        return delegate.getSettings();
    }

    @Override public final void terminateScheduler() {
        delegate.terminateScheduler();
    }

    @Override public final void terminateAfterException(Throwable x) {
        delegate.terminateAfterException(x);
    }

    @Override public final boolean tryWaitForTermination(Time timeout) {
        return delegate.tryWaitForTermination(timeout);
    }

    @Override public final int exitCode() {
        return delegate.exitCode();
    }

    @Override public final SchedulerEventBus getEventBus() {
        return delegate.getEventBus();
    }

    protected final SchedulerThreadController getDelegate() {
        return delegate;
    }
}

File: opensource/test/src/main/scala/com/sos/scheduler/engine/test/DelegatingSchedulerController.scala
package com.sos.scheduler.engine.test

import com.sos.scheduler.engine.main.SchedulerController
import org.joda.time.Duration

trait DelegatingSchedulerController extends SchedulerController {

  protected val delegate: SchedulerController

  final def terminateScheduler() {
    delegate.terminateScheduler()
  }

  final def terminateAfterException(x: Throwable) {
    delegate.terminateAfterException(x)
  }

  final def tryWaitForTermination(timeout: Duration) =
    delegate.tryWaitForTermination(timeout)

  final def exitCode =
    delegate.exitCode

  final def getEventBus =
    delegate.getEventBus
}

